pkg/distributor/distributor.go

```diff
@@ -1119,7 +1119,14 @@ func (d *Distributor) forwardSamples(ctx context.Context, userID string, ts []mi
 func (d *Distributor) Push(ctx context.Context, req *mimirpb.WriteRequest) (*mimirpb.WriteResponse, error) {
 	pushReq := push.NewParsedRequest(req)
 	pushReq.AddCleanup(func() { mimirpb.ReuseSlice(req.Timeseries) })
-	return d.PushWithMiddlewares(ctx, pushReq)
+
+	return d.GetPushFunc(nil)(ctx, pushReq)
```

Comment:
This rewrap the middlewares for each `Push()` request (previously we were not doing it). `Distributor.Push()` is called by the ruler. We should fix it.

Reply:
Done here: https://github.com/grafana/mimir/pull/3462